doc/design/resource_ownership_and_GC.md

```diff
@@ -0,0 +1,38 @@
+# Resource Ownership and GC
+
+## Resource Ownership
+
+An etcd cluster will create and manages etcd pods, services. Those resources like pods/services are owned by the etcd cluster.
+Two etcd cluster owns completely disjoint set of resources, even if they have the same name.
+For example, etcd cluster "A" was deleted and later etcd cluster "A" was created again; we think these are two different clusters.
+Thus, the new one won't be managing the resources from the old one, e.g. left as garbage.
+
+As discussed in https://github.com/coreos/etcd-operator/issues/517,
+we correlate owner (i.e. cluster) and its resources by making use of `ObjectMeta.OwnerReferences` field.
+
+For etcd pod and service, they will have only one owner -- its managing cluster.
+
+## GC
+
+GC issue: https://github.com/coreos/etcd-operator/issues/518
+
+We will talk about two GC strategies we are applying in the following.
+
+### Lazy GC
+
+When creating or deleting a cluster, remove all potential garbage via label convention:
```

Comment:
we should delete resources when deleting the cluster. this is not part of GC, but normal operations. GC only kicks in when we failed to delete the resources for some reasons.